### PR TITLE
Finished first round of recipe implementation

### PR DESCRIPTION
**Pull request by allanleal**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/thermofun/run_test.bat [hunk 1]
```diff
@@ -0,0 +1,6 @@
+cd test
+mkdir build
+cd build
+cmake .. -GNinja -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX%
```
Comment (marcelotrevisani) [inline recipes/thermofun/run_test.bat]: Please add ,``-DCMAKE_BUILD_TYPE=Release``

File: recipes/thermofun/run_test.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# if [ $CLANGXX ]; then
+#     echo 'Skipping test for osx_64 build in azure pipeline.'
+#     echo 'Reason: it seems conda`s linker doesn`t support macOS 10.14 SDK.' # @chrisburr said at gitter
+#     exit 0
+# fi
+
+cd test
+mkdir build
+cd build
+cmake .. -GNinja -DCMAKE_PREFIX_PATH=$PREFIX
```
Comment (marcelotrevisani) [inline recipes/thermofun/run_test.sh]: And also here

Comment (allanleal) [reply recipes/thermofun/run_test.sh]: Thanks @marcelotrevisani !

File: recipes/thermofun/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  # skip: True  # [win and vc<15]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake>=3.13
+    - ninja
+    - pybind11>=2.3.0
+    - nlohmann_json>=3.6.1
```
Comment (marcelotrevisani) [inline recipes/thermofun/meta.yaml]: pybind and nlohmann_json should be in host

Comment (allanleal) [reply recipes/thermofun/meta.yaml]: Thanks - just changed this as suggested!

**Comment by marcelotrevisani** [comment]
Could you please skip ``vs2008`` for now?

**Comment by allanleal** [comment]
> Could you please skip `vs2008` for now?

How do I do this?

**Comment by marcelotrevisani** [comment]
If you want to, you can skip using
```
skip: true  # [win and vc<14]
```

**Comment by marcelotrevisani** [comment]
Yeah.. the log is not helpful

File: recipes/thermofun/build.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+export CPATH=$PREFIX/include
```
Comment (marcelotrevisani) [inline recipes/thermofun/build.sh]: should you do the same for windows?

**Comment by allanleal** [comment]
> Yeah.. the log is not helpful

Indeed. There is a link error that I have no idea what is causing (and I don't have a Windows system here to investigate further).

I'll skip Windows altogether for now.

**Comment by allanleal** [comment]
I wished this was the issue! ;)
Apparently it is something more obscure. I'll fix windows compilation
issues another day.
Thanks a lot, Marcelo!

**Pull request merged**